cmd/scollector/collectors/dell_hw.go

```diff
@@ -149,7 +149,41 @@ func c_omreport_storage_pdisk(id string, md *opentsdb.MultiDataPoint) {
 		}
 		//Need to find out what the various ID formats might be
 		id := strings.Replace(fields[0], ":", "_", -1)
-		Add(md, "hw.storage.pdisk", severity(fields[1]), opentsdb.TagSet{"id": id}, metadata.Gauge, metadata.Ok, descDellHWPDisk)
+		ts := opentsdb.TagSet{"id": id}
+		Add(md, "hw.storage.pdisk", severity(fields[1]), ts, metadata.Gauge, metadata.Ok, descDellHWPDisk)
+		if len(fields) < 32 {
+			return
+		}
+		if fields[2] != "" {
```

Comment:
In theory it is parseable output. Long term I want to pull from the Dracs
on this. So I'm okay with code that is more fragile over long time. With
our integration with Opserver we will find out fast...

On Mon, Oct 5, 2015 at 11:51 AM, Craig Peterson <notifications@github.com>
wrote:

> In cmd/scollector/collectors/dell_hw.go
> <https://github.com/bosun-monitor/bosun/pull/1377#discussion_r41160647>:
>
> > @@ -149,7 +149,41 @@ func c_omreport_storage_pdisk(id string, md *opentsdb.MultiDataPoint) {
> >  		}
> >  		//Need to find out what the various ID formats might be
> >  		id := strings.Replace(fields[0], ":", "_", -1)
> > -		Add(md, "hw.storage.pdisk", severity(fields[1]), opentsdb.TagSet{"id": id}, metadata.Gauge, metadata.Ok, descDellHWPDisk)
> > +		ts := opentsdb.TagSet{"id": id}
> > +		Add(md, "hw.storage.pdisk", severity(fields[1]), ts, metadata.Gauge, metadata.Ok, descDellHWPDisk)
> > +		if len(fields) < 32 {
> > +			return
> > +		}
> > +		if fields[2] != "" {
>
> magic numbers really scare me. How confident are we in this field ordering?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bosun-monitor/bosun/pull/1377/files#r41160647>.
>